### DISCUSSION TITLE
ELOC Status

### DISCUSSION THREAD

**Discussion by LIFsCode** [LIFsCode/ELOC-3.0#54] [Ideas]
Hi @OOHehir @EDsteve , just want to synchronize with you and get your opinion about how we handle global status information in the ELOC system.

Currently multiple units (wave writer, SD card, EI, Bluetooth, etc.) have different status info which have to be shared, e.g. through BT App, LEDs.
At the moment all of these units are linked as external globally to all respective places.

I would introduce a central ELOCstatus class which is thread safe and aggregates all status Informationen. Each unit which has relevant information must add their data to ELOCstatus. That would keep each module better separated and reduce dependencies.

@OOHehir any concerns or suggestions about that?

**Comment by OOHehir** [comment]
@LIFsCode @EDsteve The only thought so far is that there would be a variety of difference types of information that may need to captured & shared, e.g.:

- SD card: mounted (bool), remaining size
- wav_writer: enabled, disabled or 'on event' (enum types or int?), sample rate, recording time, folder names
- AI/ EI: enabled, disabled (enum types or int?) & possibly a numeric value (float or int) representing the threshold at which a detection is deemed to have occurred,

I suppose templates would the natural way of handling all this..
